### PR TITLE
Deposit bitcoin

### PR DESCRIPTION
**Pull request by da-kami**
image

image

image

--- 

To be done in a follow-up PR:

- Implement actual the payment history: https://github.com/itchysats/10101/issues/165
- Implement re-loading new address once used: https://github.com/itchysats/10101/issues/166

### CODE REVIEW

File: lib/wallet/depsoit.dart [hunk 1]
```diff
@@ -0,0 +1,107 @@
+import 'package:f_logs/f_logs.dart';
+import 'package:flutter/material.dart';
+import 'package:flutter/services.dart';
+import 'package:flutter_rust_bridge/flutter_rust_bridge.dart';
+import 'package:qr_flutter/qr_flutter.dart';
+
+import '../ffi.io.dart' if (dart.library.html) 'ffi.web.dart';
```
Comment (holzeis) [inline lib/wallet/depsoit.dart:7]: nit: Can you change that to a package import?

File: lib/wallet/wallet_bitcoin.dart [hunk 2]
```diff
@@ -5,12 +5,13 @@ import 'package:ten_ten_one/cfd_trading/cfd_trading.dart';
...
 import 'package:ten_ten_one/wallet/payment_history_list_item.dart';
-import 'package:ten_ten_one/wallet/receive.dart';
 import 'package:ten_ten_one/wallet/seed.dart';
-import 'package:ten_ten_one/wallet/send.dart';
 import 'package:ten_ten_one/wallet/service_card.dart';
 import 'package:ten_ten_one/utilities/divider.dart';
 import 'package:flutter_speed_dial/flutter_speed_dial.dart';
+import 'package:ten_ten_one/wallet/withdraw.dart';
+
+import 'depsoit.dart';
```
Comment (holzeis) [inline lib/wallet/wallet_bitcoin.dart:14]: typo: dep**so**it.dart --> dep**os**it.dart

Also can you please change to a package import 🙂

File: rust/src/api.rs [hunk 3]
```diff
@@ -9,12 +9,22 @@ pub struct Balance {
     pub confirmed: u64,
 }
 
+pub struct Address {
```
Comment (holzeis) [inline rust/src/api.rs:12]: ❓ Why do we need to wrap a single attribute into a struct. Can't we just return a String in the api?

**Review by holzeis — APPROVED** [review]
LGTM 👍

File: rust/src/api.rs (hunk 3 above)
Comment (da-kami) [reply rust/src/api.rs:12]: Usually I would argue: It's not bad to wrap it in terms of extendability of the API without breaking it. If we e.g. want to add to expose a `network` field as part of the `get_address` API we can just add it without breaking changes when having an `Address` type. 

For the moment it really does not matter and returning a String would be fine - but it did not take much effort wrapping it with a type :D

**Pull request merged**